pkg/targets/adapter/cloudeventstarget/adapter_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudeventstarget
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"testing"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	adaptertest "knative.dev/eventing/pkg/adapter/v2/test"
+	loggingtesting "knative.dev/pkg/logging/testing"
+
+	fakefs "github.com/triggermesh/triggermesh/pkg/adapter/fs/fake"
+)
+
+func TestCloudEventsDispatch(t *testing.T) {
+	logger := loggingtesting.TestLogger(t)
+
+	eventRight := cloudevents.NewEvent(cloudevents.VersionV1)
+	eventRight.SetType("type")
+	eventRight.SetSource("source")
+	eventRight.SetID("id")
+	eventRight.SetTime(time.Now())
+
+	eventWrong := cloudevents.NewEvent(cloudevents.VersionV1)
+	eventWrong.SetType("unit.sendFail")
+	eventWrong.SetSource("source")
+	eventWrong.SetID("id")
+	eventWrong.SetTime(time.Now())
+
+	testCases := map[string]struct {
+		senderReady bool
+		ce          cloudevents.Event
+
+		expectedError bool
+	}{
+		"Test succeed": {
+			senderReady: true,
+			ce:          eventRight,
+
+			expectedError: false,
+		},
+		"Test failed": {
+			senderReady: true,
+			ce:          eventWrong,
+
+			expectedError: true,
+		},
```

Comment:
This is a _feature_ when using the Knative CE test client:

https://github.com/knative/eventing/blob/ec36c8637ddef2333fdc80bff8a963ffcfcc5059/pkg/adapter/v2/test/test_client.go#L67-L95

Since it lives upstream I named it `eventWrong` to make it clear that the event typed `unit.sendFail` will fail.
I will add a comment at the variable declaration. I can also replace the variable naming if you find one that sounds better.

https://github.com/triggermesh/triggermesh/pull/829/commits/dd54d8f5c9654218b753c59b6cb919825c4fb3a5

Reply:
Ahhh much clearer, I learned something!

The fact that Knative doesn't expose those as public constants made it ambiguous.